contracts/PlayFiLicenseSale.sol

```diff
@@ -264,9 +330,11 @@ IPlayFiLicenseSale
     }
 
     function _setReferral(string memory code, address receiver, uint256 commission, uint256 discount) internal {
-        referrals[code].discountPercentage == discount;
-        referrals[code].commissionPercentage == commission;
-        referrals[code].receiver == receiver;
+        if(discount > 50) revert InvalidDiscount();
+        if(commission > 50) revert InvalidCommission();
+        referrals[code].discountPercentage = discount;
+        referrals[code].commissionPercentage = commission;
+        referrals[code].receiver = receiver;
         emit referralUpdated(code, receiver, commission, discount);
```

Comment:
q: if we expect this function to be called a lot, would it make sense to make `receiver` an indexed event in order to search for it in the logs? if there's already a way to track it offchain, or if deemed not needed, feel free to just resolve

Reply:
Agree with this. Will change!

Reply:
Seems like it was already done like this in the Interface:
`event referralUpdated(string code, address indexed receiver, uint256 commission, uint256 discount);`